openssl/src/cms.rs

```diff
@@ -67,19 +67,23 @@ foreign_type_and_impl_send_sync! {
 }
 
 impl CmsContentInfoRef {
-    /// Given the sender's private key, `pkey` and the recipient's certificiate, `cert`,
+    /// Given the sender's private key, `pkey` and the (optional) recipient's certificiate, `cert`,
     /// decrypt the data in `self`.
     ///
-    /// OpenSSL documentation at [`CMS_decrypt`]
+    /// *Warning*: Not providing a certificate may leave you vulnerable to Bleichenbacher's attack on PKCS#1 v1.5 RSA padding.
+    /// See the [`OpenSSL docs`] for more information.
     ///
-    /// [`CMS_decrypt`]: https://www.openssl.org/docs/man1.1.0/crypto/CMS_decrypt.html
-    pub fn decrypt<T>(&self, pkey: &PKeyRef<T>, cert: &X509) -> Result<Vec<u8>, ErrorStack>
+    /// [`OpenSSL docs`]: https://www.openssl.org/docs/man1.1.0/crypto/CMS_decrypt.html
+    pub fn decrypt<T>(&self, pkey: &PKeyRef<T>, cert: Option<&X509>) -> Result<Vec<u8>, ErrorStack>
```

Comment:
Directly changing the signature is a breaking change. This should probably be added as a separate `decrypt2` method or something like that.

Reply:
Thanks for the review, @sfackler!

I've added a new, separate `decrypt_without_cert_check()` method to avoid a breaking change. Let me know if there's anything else you think needs to be updated 👍